heartexlabs/label-studio-frontend
# Config validation

This PR brings config validation to life.

### Proper validation of `name` and `toName` attributes.

Validations:
* Existence of `name` attribute where applicable
* Existence of `toName` attribute where applicable
* Check that `toName` references a proper tag (defined by `Validation` model)
* Check that `toName` references a tag that is present in the tree

### Proper types validation

_Notice: type validation doesn't work in development because MST performs its native typecheck when `NODE_ENV=development`._

* Validation of all built-in MST types such as enumerables, strings, numbers, etc.
* Validation of custom types (lives under `utils/customTypes`). For now `color` and `range` types are available.
  * `color` ensures that provided value is a proper CSS color (named color, HEX, RGB(a), HSL(a), etc.)
  * `range` ensures that numeric value fits the provided range. By default `range` compares value against `0..1` including ends. This type is user for `opacity` validation.

_Notice: there's now way now to check required attributes properly because of `mayBeNull` usage_

### Also done

Updated `package-lock.json` to eliminate some vulnerabilities found by `npm audit`.
